platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventIntake.java

```diff
@@ -198,31 +206,34 @@ public void addEvent(final EventImpl event) {
 
         if (prehandlePool == null) {
             // Prehandle transactions on the intake thread (i.e. this thread).
+            phaseTimer.activatePhase(EventIntakePhase.PREHANDLING);
             prehandleEvent.accept(event);
         } else {
             // Prehandle transactions on the thread pool.
             prehandlePool.submit(buildPrehandleTask(event));
         }
 
         // record the event in the hashgraph, which results in the events in consEvent reaching consensus
+        phaseTimer.activatePhase(EventIntakePhase.ADDING_TO_HASHGRAPH);
         final List<ConsensusRound> consRounds = consensusWrapper.addEvent(event, addressBook);
-        // #5762 after we calculate roundCreated, se set its value in GossipEvent so that it can be shared with other
-        // nodes
+
+        // after we calculate roundCreated, we set its value in GossipEvent so that it can be shared with other nodes
         event.getBaseEvent().setRoundCreated(event.getRoundCreated());
-        stats.addedToConsensus();
+
+        phaseTimer.activatePhase(EventIntakePhase.EVENT_ADDED_DISPATCH);
         dispatcher.eventAdded(event);
-        stats.dispatchedAdded();
+
         if (consRounds != null) {
+            phaseTimer.activatePhase(EventIntakePhase.HANDLING_CONSENSUS_ROUNDS);
             consRounds.forEach(this::handleConsensus);
-            stats.dispatchedRound();
         }
+
         if (consensus().getMinGenerationNonAncient() > minGenNonAncientBeforeAdding) {
             // consensus rounds can be null and the minNonAncient might change, this is probably because of a round
             // with no consensus events, so we check the diff in generations to look for stale events
+            phaseTimer.activatePhase(EventIntakePhase.HANDLING_STALE_EVENTS);
             handleStale(minGenNonAncientBeforeAdding);
-            stats.dispatchedStale();
         }
-        stats.doneIntakeAddEvent();
```

Comment:
I was relying on the set to `IDLE` in `addUnlinkedEvent`, but you're correct and that isn't sufficient, since self events don't follow that pathway

I'll add a set to `IDLE` here.